install/azure.go

```diff
@@ -184,3 +191,46 @@ func (k *K8sInstaller) azExec(args ...string) ([]byte, error) {
 	args = append(args, "--output", "json", "--only-show-errors")
 	return k.Exec("az", args...)
 }
+
+func (k *K8sInstaller) createAKSSecrets(ctx context.Context) error {
+	// Check if secret already exists and reuse it
+	_, err := k.client.GetSecret(ctx, k.params.Namespace, defaults.AKZSecretName, metav1.GetOptions{})
+	if err == nil {
+		k.Log("🔑 Found existing AKS secret %s", defaults.AKZSecretName)
+		return nil
+	}
+
+	var (
+		secretFileName string
+	)
+
+	ciliumVer := k.getCiliumVersion()
+	switch {
+	case versioncheck.MustCompile(">=1.12.0")(ciliumVer):
+		secretFileName = "templates/cilium-operator/secret.yaml"
+	default:
+		panic("unsupported version")
+	}
+
+	secretFile := k.manifests[secretFileName]
+
+	var secret corev1.Secret
+	err = yaml.Unmarshal([]byte(secretFile), &secret)
+	if err != nil {
+		// Developer mistake, this shouldn't happen
+		panic(err)
+	}
```

Comment:
We seem to do this (`yaml.Unmarshal` and `panic` on error) in several places. How about adding a helper function `MustUnmarshalYAML` to replace these with?